plan_delegator/src/plan_delegator.cpp

```diff
@@ -495,20 +495,11 @@ namespace plan_delegator
             if(adjusted_start_dist < original_starting_lanelet_centerline_start_point_dt){
 
                 auto previous_lanelets = wm_->getMapRoutingGraph()->previous(original_starting_lanelet, false);
-                auto previous_lanelet_to_add = previous_lanelets[0];
-
-                // pick a lanelet on the shortest path
-                for (const auto& llt : previous_lanelets)
-                {
-                    auto route = wm_->getRoute()->shortestPath();
-                    if (std::find(route.begin(), route.end(), llt) != route.end())
-                    {
-                        previous_lanelet_to_add = llt;
-                        break;
-                    }
-                }
 
                 if(!previous_lanelets.empty()){
+                    auto llt_on_route_optional = wm_->getLaneletOnShortestPath(previous_lanelets);
+                    lanelet::ConstLanelet previous_lanelet_to_add = llt_on_route_optional ? llt_on_route_optional.value() : previous_lanelets[0];
```

Comment:
Since assigning `previous_lanelets[0]` to `previous_lanelet_to_add` can in some cases result in an invalid (or impossible to generate) trajectory, could we refactor this from a ternary operator to an if/else block, and log a warning statement if a previous lanelet is not found that is on the shortest path? Let me know if you think otherwise.

This would apply to the other uses of the ternary operator in this PR as well (light controlled intersection strategic and tactical plugins, and in the other location within plan_delegator.cpp)

Reply:
I updated them to include log statement